nixos/modules/tasks/network-interfaces.nix

```diff
@@ -612,6 +657,93 @@ in
       };
     };
 
+    networking.wlanInterfaces = mkOption {
+      default = { };
+      example = {
+        wlan-adhoc0 = {
+          base = "wlp6s0";
+          type = "ibss";
+          mac = "02:00:00:00:00:01";
+        };
+      };
+      description =
+        ''
+          This option allows you to define vlan devices that tag packets
+          on top of a physical interface. The value of this option is an
+          attribute set. Each attribute specifies a vlan, with the name
+          specifying the name of the vlan interface.
```

Comment:
Currently, `networking.wireless.*` is defined in `services/networking/wpa_supplicant.nix`, which brings us to the next change that I was about to propose.

In a next step, I'll change the wpa_supplicant systemd service to that there is one process of each device (that's already marked as a FIXME in wpa_supplicant.nix).
Now, it the course of that modification, I would also like to disconnect `wpa_supplicant` from the `wireless` namespace and make it available to general interfaces, simply because with the `wired` driver `wpa_supplicant` does `802.1x` authentication if a switch requests that.

So I would prefer to move the `networking.wireless.*` stuff into `tasks/networking.nix`, rather than the other way around. If `tasks/networking.nix` gets to large at some point, we could restructure it or break it apart while switching to `systemd-networkd`? What do you think? --- Sorry for my never ending questions...